lib/apiservers/service/restapi/configure_vic_machine.go

```diff
@@ -0,0 +1,150 @@
+	})
+
+	// GET /container/version
+	api.GetVersionHandler = &handlers.VersionGet{}
+
+	// POST /container/target/{target}
+	api.PostTargetTargetHandler = operations.PostTargetTargetHandlerFunc(func(params operations.PostTargetTargetParams, principal interface{}) middleware.Responder {
+		return middleware.NotImplemented("operation .PostTargetTarget has not yet been implemented")
+	})
+
+	// GET /container/target/{target}/vch
+	api.GetTargetTargetVchHandler = &handlers.VCHListGet{}
+
+	// POST /container/target/{target}/vch
+	api.PostTargetTargetVchHandler = operations.PostTargetTargetVchHandlerFunc(func(params operations.PostTargetTargetVchParams, principal interface{}) middleware.Responder {
+		return middleware.NotImplemented("operation .PostTargetTargetVch has not yet been implemented")
+	})
+
+	// GET /container/target/{target}/vch/{vch-id}
+	api.GetTargetTargetVchVchIDHandler = &handlers.VCHGet{}
+
+	// PUT /container/target/{target}/vch/{vch-id}
+	api.PutTargetTargetVchVchIDHandler = operations.PutTargetTargetVchVchIDHandlerFunc(func(params operations.PutTargetTargetVchVchIDParams, principal interface{}) middleware.Responder {
+		return middleware.NotImplemented("operation .PutTargetTargetVchVchID has not yet been implemented")
+	})
+
+	// PATCH /container/target/{target}/vch/{vch-id}
+	api.PatchTargetTargetVchVchIDHandler = operations.PatchTargetTargetVchVchIDHandlerFunc(func(params operations.PatchTargetTargetVchVchIDParams, principal interface{}) middleware.Responder {
+		return middleware.NotImplemented("operation .PatchTargetTargetVchVchID has not yet been implemented")
+	})
+
+	// POST /container/target/{target}/vch/{vch-id}
+	api.PostTargetTargetVchVchIDHandler = operations.PostTargetTargetVchVchIDHandlerFunc(func(params operations.PostTargetTargetVchVchIDParams, principal interface{}) middleware.Responder {
+		return middleware.NotImplemented("operation .PostTargetTargetVchVchID has not yet been implemented")
+	})
+
+	// DELETE /container/target/{target}/vch/{vch-id}
+	api.DeleteTargetTargetVchVchIDHandler = operations.DeleteTargetTargetVchVchIDHandlerFunc(func(params operations.DeleteTargetTargetVchVchIDParams, principal interface{}) middleware.Responder {
+		return middleware.NotImplemented("operation .DeleteTargetTargetVchVchID has not yet been implemented")
+	})
+
+	// POST /container/target/{target}/datacenter/{datacenter}
+	api.PostTargetTargetDatacenterDatacenterHandler = operations.PostTargetTargetDatacenterDatacenterHandlerFunc(func(params operations.PostTargetTargetDatacenterDatacenterParams, principal interface{}) middleware.Responder {
+		return middleware.NotImplemented("operation .PostTargetTargetDatacenterDatacenter has not yet been implemented")
+	})
+
+	// GET /container/target/{target}/datacenter/{datacenter}/vch
+	api.GetTargetTargetDatacenterDatacenterVchHandler = &handlers.VCHDatacenterListGet{}
+
+	// POST /container/target/{target}/datacenter/{datacenter}/vch
```

Comment:
Minor, but maybe we could edit the spec such that `/container/target/{target}/datacenter/{datacenter}/vch` becomes something like `/container/target/{address}/datacenter/{name}/vch` so we end up with method names like `api.GetTargetAddressDatacenterNameVchHandler` instead of `api.GetTargetTargetDatacenterDatacenterVchHandler`.

Again, pretty minor - not worth the time spent on it if you don't see it as a long term improvement.

Reply:
I really don't like the method names, and am open to changing the spec.

I worry that including vague things like `{name}` or `{id}` in the path might limit us in the future. If we start with a path like `/foo/{name}` and then want to add a sub-resource `bar`, we can't have `/foo/{name}/bar/{name}`, so we're stuck being inconsistent (with something like `/foo/{name}/bar/{barName}`) or refactoring  (to something like `/foo/{fooName}/bar/{barName}`) and likely breaking clients using generated bindings.

To avoid painting ourselves into a corner like that, I think we'd want to qualify any generally-applicable parameters up front. (So `{address}` is probably okay as-is, but `{name}` or `{id}` should probably be qualified.)

I also think readability of the generated documentation is at least as important as readability of the generated go code, so we should keep that in mind as we consider changes.

Given that, maybe we should compare both the generated documentation and generate method names for the current style (`/foo/{foo}`) with some potential new styles like `/foo/{fooId}`?

Reply:
I agree with your concerns around `{name}` and `{id}`, especially in cases where you could supply a name or an id (datacenters/datastores for example?). I like the `/foo/{fooID}` suggestion as `fooID` can imply whatever it is you are using to identify a particular `foo`, and is not restricted to being a name or a UUID or whatever, but can be anything.